spot-vaults/contracts/BillBroker.sol

```diff
@@ -662,9 +662,13 @@ contract BillBroker is
         // Compute mint amount.
         mintAmt = valueIn.mulDiv(totalSupply_, totalReserveVal);
 
-        // Apply a combination of swap and mint fees.
+        // A single sided deposit is a combination of swap and mint.
+        // We first calculate the amount of usd swapped into perps and
+        // apply the swap fee only for that portion.
+        // The mint fees are waived, because single sided deposits
+        // push the pool back into balance.
         uint256 percOfAmtInSwapped = ONE.mulDiv(
-            usdAmtIn - mintAmt.mulDiv(s.usdBalance, totalSupply_),
+            usdAmtIn - (s.usdBalance + usdAmtIn).mulDiv(mintAmt, totalSupply_),
```

Comment:
Curious how you derived this. I get...
usdAmtIn * (1 - s.usdBalance / (s.usdBalance + s.perpBalance))

Basically applying the current asset ratio to the incoming USD amount. Maybe they're the same, but it feels more natural to use asset balances rather than LP-notes for some reason.

Reply:
hmm. You'd have to value the incoming tokens based on both the usd and perp balance (and their current prices) of the reserve. We do that do calculate `mintAmt` which is `valueIn / tvl * totalSupply`.

Now with `mintAmt`, we know what share of the reserve assets the user has a claim on.

usdClaim = mintAmt / totalSupply * (s.usdBalance + usdAmtIn)
percUsdSwapped = (usdAmtIn - usdClaim) / usdAmtIn

Reply:
I think you can factor out the prices, kind of like how any non-oraclized DEX pools works. Any normal double-sided deposit is an op that doesn't change the asset ratio of the system, so you can just look at token balances.

Reply:
I dnot think so? Prices might have diverged since the last interaction. (fmv can change for example)

Reply:
Got it. These two lines from your comment above clicked for me:
```
usdClaim = mintAmt / totalSupply * (s.usdBalance + usdAmtIn)
percUsdSwapped = (usdAmtIn - usdClaim) / usdAmtIn
```
From there, it's easy to see it in the code. If you have a way to communicate that in the code comments that's great 👍